### PR TITLE
 add pipeline.yaml support

### PR DESCRIPTION
**Pull request by Barbar1432**
refactored Pipelinedefinition file to support pipeline.yaml files as well as all related functions to it

### CODE REVIEW

File: cmd/render.go [hunk 1]
```diff
@@ -298,3 +302,15 @@ func (r *RenderCommand) printErrorOrJSON(msg string) {
 func (r *RenderCommand) printErrorOrJsonf(msg string, args ...interface{}) {
 	r.printErrorOrJSON(fmt.Sprintf(msg, args...))
 }
+
+func getPipelineDefinitionFullPath(pipelinePath string) (string, error) {
```
Comment (albertobruin) [inline cmd/render.go:306]: shouldn't this return a string slice, like 
```
[]string{"/path/to/pipeline/pipeline.yml", "/path/to/pipeline/pipeline.yaml"} 
```

File: pkg/path/walk.go [hunk 2]
```diff
@@ -27,43 +28,49 @@ func GetPipelinePaths(root, pipelineDefinitionFile string) ([]string, error) {
...
 	currentFolder := absoluteTaskPath
-	for currentFolder != filepath.VolumeName(currentFolder)+string(os.PathSeparator) && currentFolder != "/" {
-		tryPath := filepath.Join(currentFolder, pipelineDefinitionFile)
-		if _, err := os.Stat(tryPath); err == nil {
-			return currentFolder, nil
+	rootPath := filepath.VolumeName(currentFolder) + string(os.PathSeparator)
+	for currentFolder != rootPath && currentFolder != "/" {
+		for _, pipelineDefinition := range pipelineDefinitionFile {
+			tryPath := filepath.Join(currentFolder, pipelineDefinition)
+			if _, err := os.Stat(tryPath); err == nil {
```
Comment (albertobruin) [inline pkg/path/walk.go:65]: I think `os.Stat(tryPath)` might not be the perfect way to check for existence. There could be a different error from "does not exist" 
See answer here https://stackoverflow.com/questions/12518876/how-to-check-if-a-file-exists-in-go

Comment (Barbar1432) [reply pkg/path/walk.go:65]: Our existing implementation was using this line `os.Stat(tryPath)` that's why I kept it

File: cmd/render.go [hunk 3]
```diff
@@ -298,3 +302,15 @@ func (r *RenderCommand) printErrorOrJSON(msg string) {
...
+	for _, pipelineDefinitionFile := range pipelineDefinitionFile {
```
[the remark this answers is not in the document]
Comment (Barbar1432) [reply cmd/render.go]: Thanks. I had first named **`PipelinedefinitionFile`** as **pipelineDefinitionFiles**  and afterwards I have decided it to rename it to PipelineDefinitionFile. I would change it to this if that's okay that would solve the issue below as well.
 `for _, pipelineDefinition := range pipelineDefinitionFile {`

**Review by karakanb — APPROVED** [review]

**Pull request merged**